File: RetroGameGauntlet.Forms/ViewModels/ViewModelLocator.cs
namespace RetroGameGauntlet.Forms.ViewModels
{
    public static class ViewModelLocator
    {
        public static AboutViewModel AboutViewModel => RetroGameGauntletApp.Container.GetInstance<AboutViewModel>();

        public static SearchPlatformsViewModel SearchPlatformsViewModel => RetroGameGauntletApp.Container.GetInstance<SearchPlatformsViewModel>();

        public static RandomViewModel RandomViewModel => new RandomViewModel();

        public static OverviewViewModel OverviewViewModel => RetroGameGauntletApp.Container.GetInstance<OverviewViewModel>();
    }
}

File: RetroGameGauntlet.Core/ViewModels/ViewModelLocator.fs
namespace RetroGameGauntlet.Core.ViewModels

open RetroGameGauntlet.Core

/// The view model localtor. It can be used as a factory of view models.
[<AbstractClass; Sealed>]
type ViewModelLocator =

    static member AboutViewModel: AboutViewModel = 
        GauntletCore.Container.GetInstance<AboutViewModel>()

    static member RandomsViewModel: RandomsViewModel = RandomsViewModel()

    static member SearchPlatformsViewModel: SearchPlatformsViewModel =
        GauntletCore.Container.GetInstance<SearchPlatformsViewModel>()

    static member OverviewViewModel:OverviewViewModel =
        GauntletCore.Container.GetInstance<OverviewViewModel>()